core/src/main/java/nl/nn/adapterframework/jdbc/JdbcListener.java

```diff
@@ -43,11 +43,12 @@
  * @author  Gerrit van Brakel
  * @since   4.7
  */
-public class JdbcListener extends JdbcFacade implements IPullingListener {
+public class JdbcListener extends JdbcFacade implements IPullingTriggerListener {
 
 	private String startLocalTransactionQuery;
 	private String commitLocalTransactionQuery;
 	private String selectQuery;
+	private String preSelectQuery;
```

Comment:
preSelectQuery hernoemen naar peekQuery, en default hetzelfde laten zijn als selectQuery. 
Als je de query uitvoert hoef je geen resultSet.next() te doen; in plaats daarvan kan je ook !resultSet.isAfterLast() doen. Daarmee bepaal je of er een record is, maar neem je geen lock.